src/log/builder.cr

```diff
@@ -1,9 +1,17 @@
 require "weak_ref"
 
+# Used in `Log.setup` methods to configure the binding to be used.
+module Log::Configuration
```

Comment:
What's the purpose of having a separate module?

Reply:
Not exposing the builder for the setup configuration in case we want to change the implementation details in the future

Reply:
But we are actually exposing the builder... casting to `Log::Configuration` doesn't prevent using the non-exposed methods on `Log::Builder`. I'd rather avoid the complexity if it's not immediately necessary. Also the `Log::Configuration` sounds very arbitrary. `Log::Builder` describes the purpose much better.

Reply:
Honestly I don't see how using an interface, err... module, it's adding complexity. But having the builder right now is an implementation detail. It's better for the documentation.

Reply:
How is it an implementation detail, when I can create an instance and pass it to `Log.setup`?

Reply:
The custom builder can be used for advanced usage or for testing. What I'm saying it's an implementation detail is that the object you receive within `setup` is the builder itself. We could change that in the future more easily if we just document the interface it's supposed to be used within the `setup`. Also, for beginners they could just read the docs about `Log.setup` yielding a configuration object and ignoring the builder completely.

If it bothers you too much we can remove it, but we thought it could be a small detail to simplify the api.

Reply:
It bothers me and I'd not merge it like this. But when others agree, it's fine by me.

Reply:
It's either a module like this or a dsl struct to be returned that will hide the builder.

The fact that `Log.setup` receives a builder it's a side effect of testing. Although it can be used for creating isolated pool of loggers. But I would rather discourage that scenario.